### PR TITLE
Feature/improve network error handling

### PR DESCRIPTION
**Pull request by mgallien**

### CODE REVIEW

File: src/libsync/propagatedownload.cpp [hunk 1]
```diff
@@ -828,7 +829,11 @@ void PropagateDownloadFile::slotGetFinished()
                 &propagator()->_anotherSyncNeeded, errorBody);
         }
 
-        done(status, errorString);
+        if (err == QNetworkReply::RemoteHostClosedError || err == QNetworkReply::TemporaryNetworkFailureError) {
+            errorCategory = ErrorCategory::NetworkError;
```
Comment (allexzander) [inline src/libsync/propagatedownload.cpp:833]: @matthieugallien You have added a method `ErrorCategory PropagatorJob::errorCategoryFromNetworkError(QNetworkReply::NetworkError error)`. Why not use it here?

File: src/libsync/propagateremotemkdir.cpp [hunk 2]
```diff
@@ -162,10 +162,14 @@ void PropagateRemoteMkdir::finalizeMkColJob(QNetworkReply::NetworkError err, con
             job->start();
         }
     });
-    connect(propfindJob, &PropfindJob::finishedWithError, this, [this]{
+    connect(propfindJob, &PropfindJob::finishedWithError, this, [this] (QNetworkReply *reply) {
+        auto err = QNetworkReply::NetworkError::UnknownNetworkError;
+        if (reply) {
```
Comment (allexzander) [inline src/libsync/propagateremotemkdir.cpp:167]: @mgallien Can't you just go with `const auto err = reply ? reply->error() : QNetworkReply::NetworkError::UnknownNetworkError;`

**Review by allexzander — COMMENTED** [review]
@mgallien Please fix compile errors, and see my comments (nitpicks mostly). Also, I have noticed tests are failing.

File: src/gui/tray/activitylistmodel.cpp [hunk 3]
```diff
@@ -545,11 +559,47 @@ void ActivityListModel::addEntriesToActivityList(const ActivityList &activityLis
...
+    } else {
+        _durationSinceDisconnection.invalidate();
+    }
+
+    _accountStateWasConnected = _accountState->isConnected();
+}
+
+void ActivityListModel::addErrorToActivityList(const Activity &activity, ErrorType type)
+{
+    qCInfo(lcActivity) << "Error successfully added to the notification list: " << activity._message << activity._subject << type;
```
[the remark this answers is not in the document]
Comment (mgallien) [reply src/gui/tray/activitylistmodel.cpp:583]: good point

File: src/libsync/propagateremotemkdir.cpp [hunk 4]
```diff
@@ -162,10 +162,11 @@ void PropagateRemoteMkdir::finalizeMkColJob(QNetworkReply::NetworkError err, con
             job->start();
         }
     });
-    connect(propfindJob, &PropfindJob::finishedWithError, this, [this]{
+    connect(propfindJob, &PropfindJob::finishedWithError, this, [this] (QNetworkReply *reply) {
+        auto err = reply ? reply->error() : QNetworkReply::NetworkError::UnknownNetworkError;
```
Comment (allexzander) [inline src/libsync/propagateremotemkdir.cpp:166]: @mgallien `cons auto err`

**Review by allexzander — APPROVED** [review]

**Comment by allexzander** [comment]
@matthieugallien Added just one more comment, but more of a nitpick. Kindly do not forget to clean-up the history.

**Pull request merged**